system/ee/ExpressionEngine/Addons/forum/mod.forum_core.php

```diff
@@ -339,14 +339,24 @@ public function _fetch_read_topics($new_id = false)
         if ($query->num_rows() == 0) {
             $this->read_topics_exist = false;
             $read_topics = $this->_fetch_read_topics_cookie($new_id);
+            /*Malicious cookie clearing.
+            read_topics should be in the form of a topic id and a time int.
+            Value should also be recent in case someone is trying to put in incorrect time values (No future values allowed either)*/
+            $clean_read_topics = array();
+            foreach($read_topics as $key => $val){
+                if(is_int($val) && (ee()->localize->now - $val) < 300 && (ee()->localize->now - $val) > -1){
```

Comment:
@bakin1999 I'm afraid this might introduce a bug. I'm thinking we should probably reduce this down to just the `is_int($val)` part. 

Also, I was hoping you could update the `foreach` to follow PSR-12.